### PR TITLE
View granted permissions

### PR DESCRIPTION
**Pull request by anushkavidanage**
# Pull Request Details

## What issue does this PR address

- View permissions granted from other PODs

- Link to associated issue: https://github.com/anusii/solidpod/issues/199

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [ ] Chrome
  - [ ] iOS
  - [ ] Linux
  - [ ] MacOS
  - [x] Windows
- [ ] Added 2 reviewers (or 1 for private repositories then they add another)

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev

### CODE REVIEW

File: lib/src/solid/read_external_pod.dart [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  final fileExists = await checkResourceStatus(fileUrl);
+
+  if (fileExists == ResourceStatus.exist) {
+    try {
+      final fileContent = await fetchPrvFile(fileUrl);
+
+      // Decrypt if reading an encrypted file
+
+      await KeyManager.hasSharedIndividualKey(fileUrl);
```
Comment (cdawei) [inline lib/src/solid/read_external_pod.dart:66]: Looks like this line is not necessary.

File: lib/src/widgets/shared_resources_table.dart [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                                          ),
+                                          child: Text(fileContent as String),
+                                        ),
+                                      ],
+                                    ),
+                                    actions: [
+                                      TextButton(
+                                          onPressed: () {
+                                            // Close the dialog
+                                            Navigator.of(context).pop();
```
Comment (cdawei) [inline lib/src/widgets/shared_resources_table.dart:151]: I got a null pointer exception here which seems to be related to the use of `context` (when clicking the `Ok` button in the `File Content` dialog.

Replacing `context` with `ctx` seems to fix the issue.

**Review by cdawei — APPROVED** [review]
Thanks Anushka. The functionality worked in my test except few minor issues (please see the detailed comments).

**Comment by anushkavidanage** [comment]
Thanks Dawei. Fixed those now

**Review by gjwgit — APPROVED** [review]

**Pull request merged**